packages/core/src/state-management/state-manager.ts

```diff
@@ -75,14 +75,17 @@ export class StateManager {
   /**
    * Takes a stream state that might not contain the complete log (and might in fact contain only the
    * genesis commit) and kicks off the process to load and apply the most recent Tip to it.
-   * @param state$
-   * @param timeoutMillis
+   *
+   * @param state$ - Current stream state.
+   * @param timeoutMillis - How much time do we wait for a response from the network.
+   * @param hint - Tip to try while we are waiting for the network to respond.
    */
-  async sync(state$: RunningState, timeoutMillis: number): Promise<void> {
+  async sync(state$: RunningState, timeoutMillis: number, hint?: CID): Promise<void> {
     const tip$ = this.dispatcher.messageBus.queryNetwork(state$.id)
+    const tipSource$ = hint ? merge(tip$, of(hint)) : tip$
```

Comment:
Does the order of the entries within `tipSource$` matter?  Should we swap `tip$` and `of(hint)` to make sure the hint is processed first?  Or will `pipe` already handle moving on to the hint if the `tip$` promise isn't ready yet?

Reply:
The order for `merge` does not matter. We `applyTip` on a CID that is _emitted_ first, and it is most probably `of(hint)` here.